### PR TITLE
pytest version for simple-test

### PR DESCRIPTION
**Pull request by flomnes**

### CODE REVIEW

File: src/tests/end-to-end/test_simple-test.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    return op[0]
+
+def check_integrity(output : Path, expected_values : np.array):
+    assert len(expected_values) == 8
+    integrity_path = find_integrity_path(output)
+    integrity_file = open(str(integrity_path), 'r')
+    output_values = list()
+    for x in integrity_file:
+        output_values.append(float(x))
+    np.testing.assert_equal(expected_values, output_values)
```
Comment (sgatto) [inline src/tests/end-to-end/test_simple-test.py:23]: to ease the reading I propose that this support function returns the output values and that the assert is performed in the test function

```suggestion
def get_interity_check_values(output : Path) -> np.array :
    integrity_path = find_integrity_path(output)
    integrity_file = open(str(integrity_path), 'r')
    output_values = list()
    for x in integrity_file:
        output_values.append(float(x))
    assert len(output_values) == 8
    return output_values
```

File: src/tests/end-to-end/test_simple-test.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+def launch_solver(study_path, use_ortools = False, ortools_solver = "sirius"):
+    # Find solver executable
+    search_result = list()
+    for path in Path.cwd().parent.parent.parent.rglob('solver/antares-*.*-solver*'):
+        search_result.append(path)
+    # Eliminate swap version
+    solver = list(filter(lambda x: "swap" not in str(x), search_result))
+    assert len(solver) == 1
+    solver_path = str(solver[0])
```
Comment (sgatto) [inline src/tests/end-to-end/test_simple-test.py:33]: I propose to extract this part of the code to a function named `find_solver_executable` or `get_solver_exeutable`

File: src/tests/end-to-end/test_simple-test.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    # TODO check return value
+    assert "Solver returned error" not in output[0].decode('utf-8')
+
+
+def test_free_data_sample_no_ortools(tmp_path):
+    # Study copied to temporary directory
+    fds = Path.cwd().parent / 'data' / 'free_data_sample'
+    copy_tree(str(fds), str(tmp_path))
+    study_path = str(tmp_path)
```
Comment (sgatto) [inline src/tests/end-to-end/test_simple-test.py:50]: refactor: extract to support function `get_temp_study_path(origin_study_path)-> Path`

**Review by sgatto — APPROVED** [review]
OK for me, thanks!
I propose some refactoring to extract functions and improve the readibility

File: src/tests/end-to-end/test_simple-test.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    output_values = list()
+    for x in integrity_file:
+        output_values.append(float(x))
+    assert len(output_values) == 8
+    return output_values
+
+def launch_solver(study_path, use_ortools = False, ortools_solver = "sirius"):
+    # Find solver executable
+    search_result = list()
+    for path in Path.cwd().parent.parent.parent.rglob('solver/antares-*.*-solver*'):
```
Comment (guilpier-code) [inline src/tests/end-to-end/test_simple-test.py:28]: I did not understand the  'solver/antares-*.*-solver*'. What does it do exactly ?

File: src/tests/end-to-end/test_simple-test.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    expected_values = np.array([2.85657392370263e+11,
+				0.00000000000000e+00,
+                                2.85657392370263e+11,
+				2.85657392370263e+11,
+				2.85657544872729e+11,
+				0.00000000000000e+00,
+				2.85657544872729e+11,
+				2.85657544872729e+11])
+    output_path = Path(study_path) / 'output'
+    check_integrity(output_path, expected_values)
```
Comment (guilpier-code) [inline src/tests/end-to-end/test_simple-test.py:63]: Something bothers me : where is function  **check_integrity** defined ?

File: src/tests/end-to-end/test_simple-test.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import numpy as np
+
+def find_integrity_path(output_dir):
+    op = []
+    for path in Path(output_dir).rglob('checkIntegrity.txt'):
+        op.append(path)
+    assert len(op) == 1
+    return op[0]
+
+def get_interity_check_values(output : Path) -> np.array :
```
Comment (guilpier-code) [inline src/tests/end-to-end/test_simple-test.py:16]: function **get_interity_check_values(...)** should be renamed **get_integrity_check_values(...)** (a **g** is missing).
This function does not seem to be called.

Comment (flomnes) [reply src/tests/end-to-end/test_simple-test.py]: done x2

File: src/tests/end-to-end/test_simple-test.py (hunk 4 above)
Comment (flomnes) [reply src/tests/end-to-end/test_simple-test.py:27]: It looks for any file that matches this regexp. It's slightly more general than the original regex, but still matches exactly 2 files in a fresh build.

File: src/tests/end-to-end/test_simple-test.py (hunk 5 above)
Comment (flomnes) [reply src/tests/end-to-end/test_simple-test.py]: Earlier in the file

**Pull request merged**